teachers_digital_platform/crtool/src/js/components/CustomerReviewToolComponent.js

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+
+import C from "../constants"; 
+import ContentDistinctive from "./distinctives/ContentDistinctive";
+import UtilityDistinctive from "./distinctives/UtilityDistinctive";
+import QualityDistinctive from "./distinctives/QualityDistinctive";
+import EfficacyDistinctive from "./distinctives/EfficacyDistinctive";
+import SurveyPageContainer from "./pages/SurveyPageContainer";
+
+export default class CustomerReviewToolComponent extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      currentPage: localStorage.getItem(C.START_DISTINCTIVE),
+
+      contentInProgress: localStorage.getItem(C.CONTENT_STATUS),
+      qualityInProgress: localStorage.getItem(C.QUALITY_STATUS),
+      utilityInProgress: localStorage.getItem(C.UTILITY_STATUS),
+      efficacyInProgress: localStorage.getItem(C.EFFICACY_STATUS),
+
+      criterionAnswers: localStorage.getItem("criterionAnswers") === null ? {} : JSON.parse(localStorage.getItem("criterionAnswers")),
+      criterionNotes: localStorage.getItem("criterionNotes") === null ? {} : JSON.parse(localStorage.getItem("criterionNotes")),
+    };
+  }
+
+  clearLocalstorage() {
+    localStorage.clear();
+    console.log('clearLocalstorage')
+    this.distinctiveClicked(C.START_DISTINCTIVE);
+
+    this.setDistinctiveStatus(C.CONTENT_DISTINCTIVE, C.STATUS_IN_START);
+    this.setDistinctiveStatus(C.UTILITY_DISTINCTIVE, C.STATUS_IN_START);
+    this.setDistinctiveStatus(C.QUALITY_DISTINCTIVE, C.STATUS_IN_START);
+    this.setDistinctiveStatus(C.EFFICACY_DISTINCTIVE, C.STATUS_IN_START);
+
+    this.setState({criterionAnswers: {} });
+    this.setState({criterionNotes: {} });
+  }
+
+  changeCriterionNotes(distinctive, key, val) {
+    this.setCriterionNotesState(key, val);
+    this.setDistinctiveInProgress(distinctive);
+  }
+
+  setCriterionNotesState(key, val) {
+    let alteredCriterionNotes =  this.state.criterionNotes
+    alteredCriterionNotes[key] = val;
+    
+    localStorage.setItem("criterionNotes", JSON.stringify(alteredCriterionNotes));
+    this.setState({criterionNotes: JSON.parse(localStorage.getItem("criterionNotes"))})
+  }
+
+  changeCriterionAnswer(distinctive, key, val) {
+    this.setCriterionAnswersState(key, val);
+    this.setDistinctiveInProgress(distinctive);
+  }
+
+  setCriterionAnswersState(key, val) {
+    let alteredCriterionAnswers =  this.state.criterionAnswers
+    alteredCriterionAnswers[key] = val;
+    
+    localStorage.setItem("criterionAnswers", JSON.stringify(alteredCriterionAnswers));
+    this.setState({criterionAnswers: JSON.parse(localStorage.getItem("criterionAnswers"))})
+  }
+
+  setDistinctiveInProgress(changedDistinctive) {
+    this.setDistinctiveStatus(changedDistinctive, C.STATUS_IN_PROGRESS)
+  }
+
+  setDistinctiveComplete(changedDistinctive) {
+    this.setDistinctiveStatus(changedDistinctive, C.STATUS_COMPLETE)
+  }
+
+  setDistinctiveStatus(changedDistinctive, distinctiveStatus) {
```

Comment:
I'm not sure what you mean here.  I do believe the these constants will evolve over time, so personally I'm okay leaving them alone for now and letting them evolve as more of the tool is developed.  Could you please expand on what you mean here?

Reply:
To me, it looks like the `*_DISTINCTIVE` constants are used almost solely to provide input to the `switch` statement in `setDistinctiveStatus`. Once matched, though, the `*_STATUS*` constants are used to update localStorage. Instead of having these separate constants, just using the `*_STATUS` constant itself would be equivalent.

Reply:
I see what you are saying, and yes you are correct as of right now.  But if you look at SurveyPageContainer you will see it must know what the specific distinctive page is and not the status so the *_DISTINCTIVE is used there.  Also the SurveyPageContainer will have a sister ReviewPageContainer that will do the same thing for review pages.  So I'm thinking its good to have the separation. 

Thoughts?

Reply:
I think the "status" name of the constants is actually disingenuous. Both the `*_DISTINCTIVE` and `*_STATUS` constants are just used as labels, ie the `*_STATUS` constants *aren't* statuses, they are labels for statuses. From what I can tell, both these labels are actually referring to the same thing, namely, which page is currently selected. Changing these labels to something more straightforward like `PAGE` or `LABEL` I think would make the code clearer.